logos/src/logos/sdi/ollama_facade.py

```
@@ -0,0 +1,305 @@
+        2. on_request_begin_processing() - Request starts processing (increment active count)
+        3. on_request_complete() - Request finishes (decrement active count)
+
+        Args:
+            request_id: Request identifier (same as on_request_start)
+
+        Raises:
+            KeyError: If request_id not found
+        """
+        with self._lock:
+            if request_id not in self._request_tracking:
+                raise KeyError(f"Request {request_id} not found in tracking")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)